### ISSUE TITLE
Performance issues

### ISSUE THREAD

**Issue by Maudboomen** [nathan-JJRplus/EZWebsocket#4]
Dear Nathan, 

Thank you for making this widget publicly available. 
We used the widget, but we unfortunately experienced performance issues with this widget in web. 
When having many active sessions, performance issues occur. 

We looked at this with Mendix expert services and found out the following: 

The main cause is in following code snippet ( ‘javasource/ezwebsocket/WebsocketEndpoint.java’ at the bottom of the code): 
```
   private boolean validateSession(String csrfToken) {
          Iterator<? extends ISession> activeSessions = Core.getActiveSessions().iterator();
          while (activeSessions.hasNext()) {
            ISession session = activeSessions.next();
            if (session.getCsrfToken().equals(csrfToken)) {
              return true;
            }
          }
          throw new RuntimeException("Invalid csrfToken");
        }
        
```
Crucial is the call to Core.getActiveSessions(). It turns out that for each session, it performs a retrieve-by-id on the Account. When having many active sessions, this causes performance issues. I think it would be more efficient here to perform a targeted retrieve on the session. You can replace the function with this:

```
      private boolean validateSession(String csrfToken) {
            try {
                  List<Session> activeSessions = Session.load(getContext(),
                              String.format("[%s='%s']", Session.MemberNames.CSRFToken, csrfToken));
                  if (!activeSessions.isEmpty()) {
                        return true;
                  } else {
                        throw new RuntimeException("Invalid csrfToken");
                  }
            } catch (CoreException e) {
                  throw new RuntimeException(e);
            }
      }
```

Would it be possible to replace the code? 

Thank you in advance and looking forward to your reply. 

Kind regards, 

Maud van den Boomen

**Comment by nathan-JJRplus** [comment]
Hi Maud,
I will make sure there is a new version available coming monday!

Thanks for helping me improve the module 😄

**Comment by nathan-JJRplus** [comment]
Fixed in version 1.4.1
